### DISCUSSION TITLE
AttributeError: 'EmailOperator' object has no attribute 'add_edge_info'

### DISCUSSION THREAD

**Discussion by arvindeybram** [apache/airflow#22030] [General]
### Apache Airflow version

2.2.2

### What happened

I created an email operator fucntion -

```
def AlertOperator(contact_list, message, html_content, task_id=''):
    if not task_id:
        task_id = "taskFailed"
    return EmailOperator(
        trigger_rule=TriggerRule.ONE_FAILED,
        task_id=task_id,
        to=contact_list,
        subject=message,
        html_content=html_content)

```

This function is used by all my dags - 
while adding it to one of the new dags I see dag import error

`Broken DAG: [/home/platform/airflow/dags/reports/zynga.py] Traceback (most recent call last):
  File "/home/platform/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 1481, in set_downstream
    self._set_relatives(task_or_task_list, upstream=False, edge_modifier=edge_modifier)
  File "/home/platform/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 1470, in _set_relatives
    edge_modifier.add_edge_info(self.dag, self.task_id, task.task_id)
AttributeError: 'EmailOperator' object has no attribute 'add_edge_info'`
function

    taskFailed1 = fn.AlertOperator(
        custom_args["contact_list"],
        "message",
        '<h3>htmlmessage</h3>',
        'task1'
    )
    taskFailed2 = fn.AlertOperator(
        custom_args["contact_list"],
        "message",
        '<h3>htmlmessage</h3>',
        'task2'
    )
    fun1().set_downstream(taskFailed1, func2() >> taskFailed2)

### What you expected to happen

Email Operator should have worked properly

### Operating System

centos 7

### Deployment

Other

### Deployment details

I have installed airflow via pip on bare metal

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by potiuk** [comment]
There is no way anyone can  deug your code - based on the fragments that are somewhat complex and not complete. Your pieces of code do enot explain what you want to do. Solving your code problems is not something that you can get help here (but if you attempt to describe it better, you might).
